src/www/containers/settingspage/SettingsPage.jsx

```diff
@@ -0,0 +1,438 @@
+import Select from "react-select";
+import axios from "axios";
+import { connect } from "react-redux";
+import makeAnimated from "react-select/animated";
+import { withAlert } from "react-alert";
+import { withTranslation } from "react-i18next";
+
+//import { Link, Redirect } from "react-router-dom";
+//const favoutite = require("./favourite.json");
+//const categories = require("./categories.json");
+
+const options = [
+  { value: "Category1", label: "" },
+  { value: "Category2", label: "" },
+  { value: "Category3", label: "" },
+  { value: "Category4", label: "" },
+  { value: "Category5", label: "" },
+  { value: "Category6", label: "" },
+  { value: "Category7", label: "" },
+  { value: "Category8", label: "" },
+  { value: "Category9", label: "" },
+  { value: "Category10", label: "" },
+  { value: "Categorυ11", label: "" },
+  { value: "Category12", label: "" },
+  { value: "Category13", label: "" },
+  { value: "Category14", label: "" },
+  { value: "Category15", label: "" },
+  { value: "Category16", label: "" },
+  { value: "Category17", label: "" },
+  { value: "Category18", label: "" },
+  { value: "Category19", label: "" },
+  { value: "Category20", label: "" },
+];
+const favourite_options = [
+  { value: "Category1", label: "" },
+  { value: "Category2", label: "" },
+  { value: "Category3", label: "" },
+  { value: "Category4", label: "" },
+  { value: "Category5", label: "" },
+  { value: "Category6", label: "" },
+  { value: "Category7", label: "" },
+  { value: "Category8", label: "" },
+  { value: "Category9", label: "" },
+  { value: "Category10", label: "" },
+  { value: "Categorυ11", label: "" },
+  { value: "Category12", label: "" },
+  { value: "Category13", label: "" },
+  { value: "Category14", label: "" },
+  { value: "Category15", label: "" },
+  { value: "Category16", label: "" },
+  { value: "Category17", label: "" },
+  { value: "Category18", label: "" },
+  { value: "Category19", label: "" },
+  { value: "Category20", label: "" },
+];
+
+const categoriesList = require("./categories.json");
+const favouriteCategories = require("./favouriteCategories.json");
+
+const coordinates = { lat: 40.63666412, lng: 22.942162898 };
+
+const animatedComponents = makeAnimated();
+
+const mapState = (state) => ({
+  user: state.auth.user,
+  loggedIn: state.auth.loggedIn,
+});
+
+export default withAlert()(
+  connect(
+    mapState,
+    undefined
+  )(
+    withTranslation()(
+      class SettingsPage extends Component {
+        constructor(props) {
+          super(props);
+          this.state = {
+            activeTab: "profile",
+            selectedOption: [],
+            mismatch: false,
+            position: coordinates,
+            fc: "",
+            old_password: "",
+            new_password: "",
+            passwordConfirm: "",
+            username: "",
+            bio: "",
+            profilePicture: "",
+          };
+        }
+
+        formatCategories = (categories, options) => {
```

Comment:
1. I think I don't understand what you mean about "running on every call of render()". How is it running on every render? Are you referencing to the favorite categories <Select> component that refreshes every time a user adds or removes a category from his favorite categories list, or is it something else i don't understand about the way and the place I have implemented the function?

2. I changed the for loop to a .map() but I don't think I can remove the slicing part, since the procedure I follow is taking a formated const with 20 empty categories, fill it every time with the favorite-categories the user has selected ( <=20, e.g. 5 ), fill the first 5, and only return the format's first 5 categories. Any better idea to implement this would be appreciated.

Reply:
2. I changed the for loop to a .map() but I don't think I can remove the slicing part, since the procedure I follow is taking a formated const with 20 empty categories, fill it every time with the favorite-categories the user has selected ( <=20, e.g. 5 ), fill the first 5, and only return the format's first 5 categories. Any better idea to implement this would be appreciated.